# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/protocol/http/HttpChannelChild.h
# Commit: 25822279ba5b
# Full Hash: 25822279ba5b7b695bd9173969877e8d5ba19000
# Author: Junior Hsu <juhsu@mozilla.com>
# Date: 2020-06-24 09:31:07
# Regressor Bug: 1633935
# File Overlap Count: 1
# Description:
#   Bug 1633935 - P12 top-level IPC stream filter should happen before OnStartRequest, r=mayhemer,necko-reviewers
#   
#   The WebRequest will suspend the channel in nsHttpChannel::PrepareToConnect(), apply the stream filter, send the main thread IPC  and resume.
#   That is, PHttpChannel::AttachStreamFilter should be handled in Child before OnStartRequest goes to listener.
#   
# ==============================================================================

diff -r e87b11c172b9 -r 25822279ba5b netwerk/protocol/http/HttpChannelChild.h
--- a/netwerk/protocol/http/HttpChannelChild.h	Tue Jun 23 16:58:38 2020 +0000
+++ b/netwerk/protocol/http/HttpChannelChild.h	Tue Jun 23 16:58:40 2020 +0000
@@ -463,9 +463,9 @@
   // True if RecvOnStartRequestSent was received.
   uint8_t mRecvOnStartRequestSentCalled : 1;
 
-  // True if this channel is suspened by waiting for permission and cookie. That
-  // is, RecvOnStartRequestSent is received.
-  uint8_t mSuspendedByWaitingForPermissionAndCookie : 1;
+  // True if this channel is suspened by waiting for permission, cookie or
+  // stream filter. That is, RecvOnStartRequestSent is received.
+  uint8_t mSuspendedByWaitingForPermissionCookieStreamFilter : 1;
 
   void FinishInterceptedRedirect();
   void CleanupRedirectingChannel(nsresult rv);